kernel: enhance documentation of z_arch_buffer_validate

ioannisg opened the pull request:
This commit enhances the documentation of z_arch_buffer_validate
describing the cases where the validation is performed
successfully, as well as the cases where the result is
undefined.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
kernel: enhance documentation of z_arch_buffer_validate

This commit enhances the documentation of z_arch_buffer_validate
describing the cases where the validation is performed
successfully, as well as the cases where the result is
undefined.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg commented:
I 'm not so familiar, @andrewboie , with writing kernel API documentation, so I appreciate your feedback here.

andrewboie commented:
This looks fine.
It may be worthwhile to also leave a note under doc/..../syscalls.rst

andrewboie commented:
Btw there is a typo in the patch subject line 'ducumentation'

ioannisg commented:
> 
> 
> This looks fine.
> It may be worthwhile to also leave a note under doc/..../syscalls.rst

Sure, i am going to do that.

dbkinder commented on kernel/include/kernel_internal.h:
```diff
@@ -125,6 +125,18 @@ extern void z_arch_mem_domain_destroy(struct k_mem_domain *domain);
+ * supplied buffer, if the user access permissions of the entire buffer are
+ * enforced by a single, enabled memory management region.
+ *
+ * In some architectures the validation is will always return failure,
```
```suggestion
 * In some architectures the validation will always return failure
```

ioannisg replied:
Fixed now.

dbkinder reviewed (changes requested):
so close...

ioannisg committed:
kernel: userspace: correct documentation for Z_SYSCALL_MEMORY_ macros

Corrections in the documentation of arguments in
Z_SYSCALL_MEMORY, Z_SYSCALL_MEMORY_READ, and
Z_SYSCALL_MEMORY_WRITE macros.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg reviewed (commented):
Fixed the minor style issue and added one more commit, to correctly
document the arguments in Z_SYSCALL_MEMORY macros.
@dbkinder please, re-review.
